docs/qt.rst

```diff
@@ -0,0 +1,32 @@
+# Qt
+
+For Qt-based applications we have the `org.kde.Sdk` that will offer us most Qt
+modules and KDE Frameworks for our applications to use.
+
+For example, the following JSON makes building a random Qt application really
+straight-forward.
+
+.. code-block:: json
+
+    {
+      "id": "org.kde.blinken",
+      "branch": "master",
+      "runtime": "org.kde.Platform",
+      "runtime-version": "5.9",
+      "sdk": "org.kde.Sdk",
+      "command": "blinken",
+      "finish-args": ["--share=ipc", "--socket=x11", "--socket=wayland" ],
```

Comment:
`--device=dri`?

Reply:
And `--socket=wayland` only when tested?

Reply:
You need --socket=wayland enabled, otherwise users running on Wayland won't be able to run their app as both Plasma and Gnome require Qt Wayland plugin on wayland.

Reply:
But afaik there are apps that don't support wayland and crash if this enabled.

Reply:
The --socket=wayland just allows the app to access the wayland socket outside of it's sandbox, it shouldn't make the app crash or force it to use wayland in an X11 environment.

Reply:
> And --socket=wayland only when tested?
For that matter xcb needs testing too. It's not like xcb always works.
Will include a comment about dri.

Reply:
> The --socket=wayland just allows the app to access the wayland socket outside of it's sandbox, it shouldn't make the app crash or force it to use wayland in an X11 environment.

Kdenlive does crash when trying to run it under wayland natively:
```
$ flatpak run --socket=wayland org.kde.kdenlive
QSocketNotifier: Can only be used with threads started with QThread
Using Wayland-EGL
Using the 'xdg-shell-v6' shell integration
trying to load "/usr/lib/plugins/kf5/kio/file.so" from "/usr/lib/plugins/kf5/kio/file.so"
Attempting to create QWindow-based QOffscreenSurface outside the gui thread. Expect failures.
Attempting to create QWindow-based QOffscreenSurface outside the gui thread. Expect failures.
ASSERT failure in QCoreApplication::sendEvent: "Cannot send events to objects owned by a different thread. Current thread 0x0x2708aa0. Receiver 'QOffscreenSurface' (of type 'QWindow') was created in thread 0x0x4251b30", file kernel/qcoreapplication.cpp, line 576
```

Reply:
Then kdenlive needs fixing.